### DISCUSSION TITLE
Break out room 12 - Practical series 3

### DISCUSSION THREAD

**Discussion by Rawizard** [IBAHCM/RPiR#212] [Archive]
Break out room 12 - Practical series 3 group members:

Jamie crowther - @Rawizard  
David Anderson - @davidanderson102
Sophie Zawadzki - @SophieZawadzki
Daren stuart - @2032863s

**Comment by Rawizard** [comment]
My practical series 3 repository: https://github.com/IBAHCM/Rawizardseries03.git

**Reply by Rawizard** [reply]
Repos commented on: 

https://github.com/IBAHCM/DavidAndersonSeries03.git

https://github.com/IBAHCM/SophieZawadzkiSeries03.git

**Reply by SophieZawadzki** [reply]
thanks for the comments guys! really helpfu;!

**Comment by SophieZawadzki** [comment]
My practical 3 repository (work in progress lol): https://github.com/IBAHCM/SophieZawadzkiSeries03.git

**Reply by Rawizard** [reply]
Hey Sophie, just a heads up link doesn't seem to work - "404 error".

**Reply by 2032863s** [reply]
hey guys, think the link might be just missing the '.git' at the end

**Reply by SophieZawadzki** [reply]
thanks guys, just edited it now!

**Comment by 2032863s** [comment]
my repository: 
https://github.com/IBAHCM/darrenstuartseries03.git

**Reply by DanniAnderson** [reply]
Hey Darren, I can't access your repo.  Can you double check I have access

**Reply by Rawizard** [reply]
Yeah Daren, same thing for me.

**Reply by 2032863s** [reply]
oh dear lord, apologies lads. I had some technical problems and couldn't access the local files for the first repo so was scrambling around trying to get everything sorted. better late than never. https://github.com/IBAHCM/darrenstuartseries3.git

**Reply by 2032863s** [reply]
its been a nightmare haha

**Reply by SophieZawadzki** [reply]
I'll look yours over when we break up Darren!

**Comment by DanniAnderson** [comment]
Hey, here's mine

https://github.com/IBAHCM/DavidAndersonSeries03.git

**Comment by DanniAnderson** [comment]
Btw guys I think we still need to give each other access to the repo's cause I cant open any of yours

**Comment by Rawizard** [comment]
Hey,


I am in the same boat, i just went on to my repo and settings > manage access and added the RPiR course, can you know access mine?

J

On 4 Feb 2021, at 13:07, David Anderson <notifications@github.com<mailto:notifications@github.com>> wrote:



Btw guys I think we still need to give each other access to the repo's cause I cant open any of yours

**Reply by DanniAnderson** [reply]
Yeah, I got yours now

**Comment by Rawizard** [comment]
Sweat!

On 4 Feb 2021, at 13:13, David Anderson <notifications@github.com<mailto:notifications@github.com>> wrote:



Yeah, I got yours now

**Comment by 2032863s** [comment]
sorry i couldn't make it today guys, going to try finish off 3-2 tonight. Is anyone else receiving the dreaded 'object of type closure is not subsettable' error message after returning the list in the function and trying to run the demo?

**Reply by SophieZawadzki** [reply]
I got something similar, "Error in !data$end.experiment : invalid argument type", did you work out how to fix it?

**Reply by 2032863s** [reply]
Try removing the last chunk of code in the function where you're checking for global variables and replace it with " RPiR:: assert_no_globals (your_function_name). Document and install with devtools, then try running again. Richard recommended doing that for me and it seemed to solve a lot of issues i was having. Hope that helps :)

**Reply by SophieZawadzki** [reply]
thank you! ill give it a go!

**Reply by 2032863s** [reply]
I had a look at the code for your function, i think try amending it so it looks like this:

  effective.birth.rate <- birth.rate * timestep
  effective.death.rate <- death.rate * timestep

 ((effective.birth.rate >= 1) || (effective.death.rate >= 1))
    stop("Effective rate too high, timestep must be too big") 

rather than having it at the bottom of the function, let me know how you get on

**Reply by SophieZawadzki** [reply]
it worked! thank you Darren! <3

**Reply by 2032863s** [reply]
left a comment in your 0304 run script sophie, hopefully that helps

**Comment by SophieZawadzki** [comment]
Hey guys, is anyone else getting an "Error in !data$end.experiment : invalid argument type" error message when trying to run the 3-2 demo? this is when using the code Richard gave us

**Comment by Rawizard** [comment]
Ah no that sucks! Sadly we still don't have access, if you go to your repository on GitHub and go on settings > manage access then add the class or us individually we will be bale to check over your work!

On 10 Feb 2021, at 11:53, Darren Stuart <notifications@github.com<mailto:notifications@github.com>> wrote:



oh dear lord, apologies lads. I had some technical problems and couldn't access the local files for the first repo so was scrambling around trying to get everything sorted. better late than never. https://github.com/IBAHCM/darrenstuartseries3.git

**Reply by 2032863s** [reply]
done
